From 7c44b0814fbdfda7f688814a98a241701c644745 Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Wed, 21 Jun 2017 16:43:06 +0800
Subject: [PATCH 1985/5242] MLK-15133-02 driver: clk: Skip enable non-critical
 clks on imx8mq

commit  237e64cdf90ac856c266a0ce68b7bc49293a4637 from
https://source.codeaurora.org/external/imx/linux-imx.git

Only enable the system critical clks by default, other clks only
need to be enabled when it is used by the driver.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8mq.c |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/imx/clk-imx8mq.c b/drivers/clk/imx/clk-imx8mq.c
index d2e101a..6e1761c 100644
--- a/drivers/clk/imx/clk-imx8mq.c
+++ b/drivers/clk/imx/clk-imx8mq.c
@@ -262,6 +262,14 @@
 static const char *imx8mq_ecspi3_sels[] = {"osc_25m", "sys2_pll_200m", "sys1_pll_40m", "sys1_pll_160m",
 					   "sys1_pll_800m", "sys3_pll2_out", "sys2_pll_250m", "audio_pll2_out", };
 
+
+static int const clks_init_on[] __initconst = {
+	IMX8MQ_DRAM_PLL_OUT, IMX8MQ_CLK_AHB_CG,
+	IMX8MQ_CLK_NOC_CG, IMX8MQ_CLK_NOC_APB_CG,
+	IMX8MQ_CLK_USB_BUS_CG, IMX8MQ_CLK_NAND_USDHC_BUS_CG,
+	IMX8MQ_CLK_MAIN_AXI_CG, IMX8MQ_CLK_A53_CG,
+};
+
 static struct clk_onecell_data clk_data;
 
 static void __init imx8mq_clocks_init(struct device_node *ccm_node)
@@ -771,8 +779,8 @@ static void __init imx8mq_clocks_init(struct device_node *ccm_node)
 	of_clk_add_provider(np, of_clk_src_onecell_get, &clk_data);
 
 	/* enable all the clocks just for bringup */
-	for (i = 0; i < IMX8MQ_CLK_END; i++)
-		clk_prepare_enable(clks[i]);
+	for (i = 0; i < ARRAY_SIZE(clks_init_on);  i++)
+		clk_prepare_enable(clks[clks_init_on[i]]);
 
 	clk_set_parent(clks[IMX8MQ_CLK_AHB_SRC], clks[IMX8MQ_SYS1_PLL_133M]);
 	clk_set_parent(clks[IMX8MQ_CLK_AUDIO_AHB_SRC], clks[IMX8MQ_SYS2_PLL_1000M]);
-- 
1.7.9.5

